tests/chain.js

```diff
@@ -18,10 +18,41 @@ metasync.for([1, 2, 3, 4]).filter((item, cb) => {
 }).reduce((a, b, cb) => {
   process.nextTick(cb, null, a + b);
 }).then((result) => {
-  console.log('Chaining test done: ' + result);
-  assert.strictEqual(result, 12);  // 2 * 2 + 4 * 2
+  console.log('Chaining test #1 with process.nextTick done: ' + result);
+  assert.strictEqual(result, 12); // 2 * 2 + 4 * 2
 }).catch((error) => {
   const description = error.stack || 'Error: ' + error.toString();
   console.error(description);
   process.exit(1);
 });
+
+metasync.for([1, 2, 3, 4]).filter((item, cb) => {
+  cb(item % 2 === 0);
+}).map((item, cb) => {
+  cb(null, item * 2);
+}).reduce((a, b, cb) => {
+  cb(null, a + b);
+}).then((result) => {
+  console.log('Chaining test #2 with callbacks done: ' + result);
+  assert.strictEqual(result, 12); // 2 * 2 + 4 * 2
+}).catch((error) => {
+  const description = error.stack || 'Error: ' + error.toString();
+  console.error(description);
+  process.exit(1);
+});
+
+//TODO: check implementation, .then() should not be called on error
+metasync.for([1, 2, 3, 4]).filter((item, cb) => {
+  cb(item % 2 === 0);
+}).map((item, cb) => {
+  cb(new Error('Shit!'));
```

Comment:
I don't think it is appropriate to use such wording in an open-source project. It will surely put off some people who might have been potential contributors or even users of the library.